extension/src/test/SharedAccessSignature.test.ts

```diff
@@ -0,0 +1,46 @@
+import * as assert from "assert";
+import { SharedAccessSignature } from "../externalresources/SharedAccessSignature";
+import * as BaseAppTranslationFiles from "../externalresources/BaseAppTranslationFiles";
+
+suite("SASToken tests", function () {
+  const expiredToken =
+    "sv=2019-12-12&ss=f&srt=o&sp=r&se=2021-11-25T05:28:10Z&st=2020-11-24T21:28:10Z&spr=https&sig=JP3RwQVCZBo16vJCznojVIMvPOHgnDuH937ppzPmEqQ%3D";
+  const expiredSASToken = new SharedAccessSignature(expiredToken);
+  const currentToken = new SharedAccessSignature(
+    BaseAppTranslationFiles.BlobContainerSettings.sasToken
+  );
+  test("Current SAS Token Test", function () {
+    /**
+     * Assert that the current SASToken for the blob storage is not about to expire.
+     */
+    const daysUntilExpiration = currentToken.daysUntilExpiration();
+    if (daysUntilExpiration <= 30) {
```

Comment:
I think you could change the day constants to 180+ 90 instead of 30 + 10. It does not matter if we create a token a bit too early